packages/react-router/lib/components.tsx

```diff
@@ -253,7 +253,8 @@ export function Routes({
   children,
   location,
 }: RoutesProps): React.ReactElement | null {
-  return useRoutes(createRoutesFromChildren(children), location);
+  let routes = React.useMemo(() => createRoutesFromChildren(children), [children]);
```

Comment:
I'm 99.9% certain that `children` is always going to be changing (unless the state change happens inside of `Routes`, but it isn't stateful, so that never happens).

https://codesandbox.io/s/friendly-lehmann-zg3m65?file=/src/App.js

Reply:
It's only changing when parent component rerenders, while `useLocation` reacts to every URL change

Reply:
I mean, `useRoutes` has `useLocation` inside which makes this statement not completely true:

> it isn't stateful, so that never happens

Reply:
Children is stable when a route's own state changes. If the state change comes from above (like useLocation) the children are not stable.